src/libfetchers/input-accessor.cc

```diff
@@ -3,31 +3,6 @@
 
 namespace nix {
 
-StorePath InputAccessor::fetchToStore(
```

Comment:
Just to note but some of this code is here to minimize the diff with lazy-trees. So if we start changing this, it will increases the lazy-trees diff again.

Reply:
@edolstra happy to fix that merge conflict :)

Reply:
Right now there is a conflict in lazy trees from https://github.com/NixOS/nix/pull/8848 that is very domain-specific on both side. I can't help with that so much, but I am happy to do all the other conflicts!

Reply:
`fetchToStore` stays non-`virtual` on `lazy-trees` so I think there is no issue here. This PR makes `Store` able to consume `SourceAccessors` directly so we just use that. Anything in an `InputAcessor` beyond the underlying `SourceAccessor` `Store::addToStore` won't care about anyways.

Reply:
For the `fingerprinting`, Sure that method could go back, but we could also flip around the logic so the `SourceAccesor` itself has a `maybeLstat`-like method to content-address file system object, and the `InputAccesor` implementations do the `getCache()` to look this up.

Reply:
On lazy-trees, `InputAccessor::fetchToStore()` implements a SQLite cache for `SourcePath` -> `StorePath` mappings that I want to backport to master. It optimizes stuff like `src = <some big tree>`. That's why it's important not to get rid of `fetchToStore()`.

Reply:
OK I'll put it back

Reply:
It is now put back (not sure why this thread isn't marked outdated).

Reply:
Not on a red or green line, is probably why.